crates/client/src/client.rs

```diff
@@ -0,0 +1,167 @@
+use {
+    crate::{
+        Config,
+        CoordinatorConnection,
+        Error,
+        LoadBalancer,
+        NodeData,
+        Request,
+        RequestExecutor,
+        cluster,
+    },
+    arc_swap::ArcSwap,
+    std::{sync::Arc, time::Duration},
+    tokio::sync::oneshot,
+    wc::future::FutureExt,
+    wcn_storage_api::{StorageApi, operation as op},
+};
+
+pub struct Client<D: NodeData = ()> {
+    cluster: cluster::Cluster<D>,
+    connection_timeout: Duration,
+    _shutdown_tx: oneshot::Sender<()>,
+}
+
+impl<D> Client<D>
+where
+    D: NodeData,
+{
+    pub(crate) async fn new(config: Config) -> Result<Self, Error> {
+        let cluster_api =
+            wcn_cluster_api::rpc::ClusterApi::new().with_rpc_timeout(Duration::from_secs(5));
+
+        let cluster_api_client_cfg = wcn_rpc::client::Config {
+            keypair: config.keypair.clone(),
+            connection_timeout: config.connection_timeout,
+            reconnect_interval: config.reconnect_interval,
+            max_concurrent_rpcs: 50,
+            max_idle_connection_timeout: config.max_idle_connection_timeout,
+            priority: wcn_rpc::transport::Priority::High,
+        };
+
+        let cluster_api_client = wcn_rpc::client::Client::new(cluster_api_client_cfg, cluster_api)?;
+
+        let coordinator_api =
+            wcn_storage_api::rpc::CoordinatorApi::new().with_rpc_timeout(Duration::from_secs(2));
+
+        let coordinator_api_client_cfg = wcn_rpc::client::Config {
+            keypair: config.keypair,
+            connection_timeout: config.connection_timeout,
+            reconnect_interval: config.reconnect_interval,
+            max_concurrent_rpcs: config.max_concurrent_rpcs,
+            max_idle_connection_timeout: config.max_idle_connection_timeout,
+            priority: wcn_rpc::transport::Priority::High,
+        };
+
+        let coordinator_api_client =
+            wcn_rpc::client::Client::new(coordinator_api_client_cfg, coordinator_api)?;
+
+        // Initialize the client using one or more bootstrap nodes:
```

Comment:
nit: there's no notion of bootstrap nodes now, please remove as it can be confusing for operators